grafana/grafana
# Logs: refactor: remove unused param "index"

In explore, when you run a logs-producing Loki query, click on a log-row in the results, and click on the [+] icon by a label, it will modify the query to produce only log-lines with that label.
if you have multiple queries in explore at the same time, this will modify all the queries.

the code that handles this functionality also accepted an optional `index`, which caused that not all queries were modified.
i've found out that this param is never used, so i removed the code that handles it.

(i will modify the code involved here to handle https://github.com/grafana/grafana/issues/56314 , but this refactor seemed like a small self-containable PR)

how to test:
1. `make devenv sources=loki`
2. go to explore, add two loki queries, in both say `{source="data"}`
3. open a log-row, click on the [+] by any label, and verify that both loki-queries were adjusted
